.github/workflows/test-and-publish-nightly.yml

```diff
@@ -51,64 +51,100 @@ jobs:
             java-version: 11
             java-distribution: corretto
             container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
+          - runson:
+              display: amazonlinux-2023 # Use dash instead of colon, because GHA artifacts cannot contain colons
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            container: amazonlinux:2023
+          - runson:
+              display: amazonlinux-2
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            # Make sure this is in sync with java-version/distribution for UX purposes
+            java-package: java-11-amazon-corretto
+            container: amazonlinux:2
     runs-on: ${{ matrix.runson.name }}
     container:
       image: ${{ matrix.container }}
       options: --privileged
     name: "test (${{ matrix.runson.display }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }})"
     steps:
+      - name: Setup
+        run: |
+          if [[ "${{ matrix.container }}" == "amazonlinux"* ]]; then
+              yum update -y
+              yum install -y make gcc-c++ libstdc++-devel libstdc++-static util-linux tar git patchelf
+          fi
       - name: Setup Java
         uses: actions/setup-java@v4
+        if: matrix.runson.display != 'amazonlinux-2'
         with:
           distribution: ${{ matrix.java-distribution }}
           java-version: ${{ matrix.java-version }}
           architecture: ${{ matrix.architecture }}
+      - name: Setup Java on amazonlinux-2
+        if: matrix.runson.display == 'amazonlinux-2'
+        run: yum install -y ${{ matrix.java-package }}
       - name: Checkout sources
+        if: matrix.runson.display != 'amazonlinux-2'
         uses: actions/checkout@v4
+      - name: Checkout sources on amazonlinux-2
+        if: matrix.runson.display == 'amazonlinux-2'
+        run: |
+          git clone https://${{ github.token }}@github.com/${{ github.repository }}.git .
       - name: Build and test
         id: build
         run: |
           HASH=$(echo ${{ github.sha }} | cut -c-7)
-          case "${{ matrix.runson.name }}" in
+          case "${{ matrix.runson.display }}" in
```

Comment:
Why do we need to switch to `display` from `name` in all these places?

Reply:
`name` is the name of the runner. For e.g. `alpine`, `amazonlinux`, and other entries relying on Docker containers to run on a specific OS, we're gonna have something like:
```
           -runson:
              display: amazonlinux-2
              name: ubuntu-latest
            ...
            container: amazonlinux:2
```